src/breadcrumb.rs

```diff
@@ -28,8 +28,12 @@ impl Breadcrumb {
     /// Creates a new Sentry breadcrumb.
     #[must_use]
     pub fn new(r#type: Option<SentryString>, message: Option<SentryString>) -> Self {
-        let type_ = r#type.map_or(ptr::null(), |type_| CString::from(type_).as_ptr());
-        let message = message.map_or(ptr::null(), |type_| CString::from(type_).as_ptr());
+        let type_ = r#type
+            .as_ref()
+            .map_or(ptr::null(), |ty| ty.as_cstr().as_ptr());
+        let message = message
+            .as_ref()
+            .map_or(ptr::null(), |msg| msg.as_cstr().as_ptr());
```

Comment:
Right ... I was suprised when this worked and Rust didn't say anything, thank you for fixing this, I will have to check the other pieces of code that are like that.

Reply:
It took me a really long time to track this down because I as far as sentry was reporting, the event was being sent successfully, and looking at the dashboard it wasn't reporting any problems processing events, but it obviously did have problems because the events just went into a black hole, it was only when I dumped the debug representation in `set_before_send` that I knew where to look.

Reply:
Yeah, amazing work, thanks!
I'm actually planning to add integration tests that query `sentry.io` itself and check if events arrive correctly.

I was wondering if this works too:
```suggestion
        let type_ = r#type
            .map_or(ptr::null(), |ty| ty.as_cstr().as_ptr());
        let message = message
            .map_or(ptr::null(), |msg| msg.as_cstr().as_ptr());
```

Reply:
Nope, that's actually the first change I did, but `.map_or` still consumes the option so the same problem occurs.

Reply:
This is terrible, well I found it at another position, fixed it, thank you.